redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package org.springframework.cloud.stream.module.redis;
+
+import java.net.UnknownHostException;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import redis.clients.jedis.Jedis;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+
+/**
+ * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
+ *
+ * @author Eric Bottard
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
+@EnableConfigurationProperties
+public class CustomRedisConfiguration extends RedisAutoConfiguration {
```

Comment:
Then maybe this concern should be brought to boot itself (cc @dsyer):
How do I easily create two instances of a whole autoconfiguration, with the second (the non-default one, so to speak) using
* another prefix for `@ConfigurationProperties`
* some way for me to make sure I inject the v2 created beans

I understand this sounds at odds with the concept of AutoConfiguration, but the problem will come up over and over in s-c-s if both the runtime and modules are based off boot